src/librustc_typeck/diagnostics.rs

```diff
@@ -1118,6 +1118,36 @@ fn main() {
 ```
 "##,
 
+E0102: r##"
+You hit this error because the compiler the compiler lacks information
+to determine a type for this variable. Erroneous code example:
+
+```
+fn main() {
+    let x: &_; // error: cannot determine a type for this local variable
+}
+```
+
+You have two possibilities to solve this situation:
+ * Give an explicit definition of the variable
+ * Infer the variable
```

Comment:
Inferring a variable by passing it to a context with a defined type is what I (and @Manishearth) had in mind.